universal-application-tool-0.0.1/app/services/question/QuestionDefinition.java

```diff
@@ -0,0 +1,85 @@
+package services.question;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.Locale;
+
+/** Defines a single question. */
+public class QuestionDefinition {
+  private final String id;
+  private final String version;
+  private final String name;
+  private final String path;
+  private final String description;
+  private final ImmutableMap<Locale, String> questionText;
+  private final ImmutableMap<Locale, String> questionHelpText;
+
+  public QuestionDefinition(
+      String id,
+      String version,
+      String name,
+      String path,
+      String description,
+      ImmutableMap<Locale, String> questionText,
+      ImmutableMap<Locale, String> questionHelpText) {
+    this.id = checkNotNull(id);
+    this.version = checkNotNull(version);
+    this.name = checkNotNull(name);
+    this.path = checkNotNull(path);
+    this.description = checkNotNull(description);
+    this.questionText = checkNotNull(questionText);
+    this.questionHelpText = checkNotNull(questionHelpText);
```

Comment:
issue: I think questionHelpText should be nullable. Some questions may not require additional text beyond the questionText.

Reply:
That's a good point - would it be null, or just an empty map?

Reply:
Suggestion: an `Optional<ImmutableMap<yada, yada>>` would make it explicit whether the help text was intentionally left blank or simply hadn't been added yet.